### PR TITLE
DRILL-8174: Convert Avro format to EVF2

### PR DESCRIPTION
**Pull request by luocooong**
# [DRILL-8174](https://issues.apache.org/jira/browse/DRILL-8174): Convert Avro format to EVF2

## Description
Second refactor of the Avro format.

## Documentation
N/A

## Testing
Use the CI.

### CODE REVIEW

**Review by cgivre — CHANGES_REQUESTED** [review]
One minor comment, but other than that, looks good.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/avro/AvroBatchReader.java [hunk 1]
```diff
@@ -38,59 +45,52 @@
...
-
-public class AvroBatchReader implements ManagedReader<FileScanFramework.FileSchemaNegotiator> {
+public class AvroBatchReader implements ManagedReader {
   private static final Logger logger = LoggerFactory.getLogger(AvroBatchReader.class);
 
-  private Path filePath;
-  private long endPosition;
+  private final Path filePath;
+  private final long endPosition;
   private DataFileReader<GenericRecord> reader;
```
Comment (paul-rogers) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/avro/AvroBatchReader.java:53]: Can `reader` be final as well?

**Review by paul-rogers — COMMENTED** [review]
The conversion looks good. Thanks for doing the work!

**Review by cgivre — APPROVED** [review]
LGTM +1

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/avro/AvroBatchReader.java [hunk 2]
```diff
@@ -122,60 +112,35 @@ public String toString() {
     } catch (IOException e) {
       logger.trace("Unable to obtain Avro reader position: {}", e.getMessage(), e);
     }
-    return "AvroBatchReader[File=" + filePath
-      + ", Position=" + currentPosition
-      + "]";
+    return String.format("AvroBatchReader [ File = %s, Position = %d ]", filePath, currentPosition);
```
[the remark this answers is not in the document]
Comment (luocooong) [reply exec/java-exec/src/main/java/org/apache/drill/exec/store/avro/AvroBatchReader.java]: Done, thanks.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/avro/AvroBatchReader.java (hunk 1 above)
Comment (luocooong) [reply exec/java-exec/src/main/java/org/apache/drill/exec/store/avro/AvroBatchReader.java]: Done, thanks.

**Pull request merged**